test/e2e/aadpodidentity_test.go

```diff
@@ -102,9 +110,18 @@ var _ = AfterSuite(func() {
 })
 
 var _ = Describe("Kubernetes cluster using aad-pod-identity", func() {
+	BeforeEach(func() {
+		testIndex++
+		fmt.Printf("\n\n[%d] %s\n", testIndex, CurrentGinkgoTestDescription().TestText)
+	})
+
 	AfterEach(func() {
 		fmt.Println("\nTearing down the test environment...")
-
+		if CurrentGinkgoTestDescription().Failed {
```

Comment:
I don't we should skip cleanup if there is a failure. Instead we should dump the logs here. If we skip clean up, it can cause issues in automation.

Reply:
Yes I agree. This is stop gap before going to the full copy logs and events atleast until that PR is done.